src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs

```diff
@@ -201,13 +201,20 @@ public static SyntaxNode DeserializeFrom(Stream stream, CancellationToken cancel
 
             if (!stream.CanRead)
             {
-                throw new InvalidOperationException(CSharpResources.TheStreamCannotBeReadFrom);
+                throw new InvalidOperationException(CodeAnalysisResources.TheStreamCannotBeReadFrom);
             }
 
-            using (var reader = new ObjectReader(stream, defaultData: GetDefaultObjectReaderData(), binder: s_defaultBinder))
+            try
             {
-                var root = (Syntax.InternalSyntax.CSharpSyntaxNode)reader.ReadValue();
-                return root.CreateRed();
+                using (var reader = new ObjectReader(stream, defaultData: GetDefaultObjectReaderData(), binder: s_defaultBinder))
+                {
+                    var root = (Syntax.InternalSyntax.CSharpSyntaxNode)reader.ReadValue();
+                    return root.CreateRed();
+                }
+            }
+            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
```

Comment:
This means we'll crash any consumers of the API that aren't VS, even if it's the consumer's mistake? I feel like we had some flag on this in the past to differentiate the two cases?